File: aos/testing/path.cc
#include "aos/testing/path.h"

#include "absl/strings/str_cat.h"

namespace aos::testing {

// Returns the path to the provided artifact which works when built both as an
// external target and in the repo.
std::string ArtifactPath(std::string_view path) {
  return absl::StrCat("../" AOS_REPO_NAME "/", path);
}

}  // namespace aos::testing

File: aos/testing/path.rs
use std::path::{Path, PathBuf};

/// Returns the correct path for a data file within AOS, accounting for whether AOS is
/// imported as an external repository.
pub fn artifact_path(path: &Path) -> PathBuf {
    Path::new("..")
        .join(Path::new(env!("AOS_REPO_NAME")))
        .join(path)
}
